packages/graphql/tests/tck/advanced-filtering.test.ts

```diff
@@ -920,7 +934,14 @@ describe("Cypher Advanced Filtering", () => {
 
                 expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
                     "MATCH (this:\`Movie\`)
-                    WHERE size([(this)-[this0:IN_GENRE]->(this1:\`Genre\`) WHERE NOT this1.name = $param0 | 1]) = 0
+                    CALL {
```

Comment:
do we need to check for at least one occurrence as well?

```cypher
MATCH (this:`Movie`)
CALL {
    WITH this
    MATCH (this)-[this0:IN_GENRE]->(this1:`Genre`)
    WHERE NOT (this1.name = $param0)
    RETURN count(this0) AS var2
}
CALL {
    WITH this
    MATCH (this)-[this0:IN_GENRE]->(this1:`Genre`)
    WHERE this1.name = $param0
    RETURN count(this0) AS var3
}
WITH *
WHERE var2 = 0 and var3  > 0
RETURN this { .actorCount } AS this"
```

Reply:
Hi @Andy2003! I'm not sure I follow. Why would we also need to check for more than one occurrence?

Reply:
Just b/c it is also done for retaltions:

https://github.com/neo4j/graphql/blob/d6837b84275f7bd69c59e7671cf5e44ffa302aed/packages/graphql/tests/tck/advanced-filtering.test.ts#L772-L782

Reply:
This has always been the behaviour for connections. However, I agree we may want to keep consistent with relationships so I have created the following issue to capture this #2713